hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/crypto/OpensslCipher.c

```diff
@@ -35,8 +35,14 @@ static void (*dlsym_EVP_CIPHER_CTX_init)(EVP_CIPHER_CTX *);
 #endif
 static int (*dlsym_EVP_CIPHER_CTX_set_padding)(EVP_CIPHER_CTX *, int);
 static int (*dlsym_EVP_CIPHER_CTX_test_flags)(const EVP_CIPHER_CTX *, int);
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
```

Comment:
1. how about adding a comment to say "the names were changed so the probes have to change their type.
2. do the actual typedefs need to be made exclusive? can see that if the actual typedef name was being recycled, but here they are being given new names

Reply:
1. addressed in f5c9b44f
2. wasn't sure which name pops up in `hadoop checknative`in case hadoop-common was compiled against OpenSSL 3.x and run with OpenSSL 1.x. Given that the symbols have same signature, the patch could be restricted to only `Java_org_apache_hadoop_crypto_OpensslCipher_initIDs`. Let me verify that...

Reply:
addressed 2. in 6c17fcfe